tools/metilene/metilene.xml

```diff
@@ -0,0 +1,158 @@
+<?xml version="1.0" encoding="utf-8"?>
+<tool id="metilene" name="metilene" version="0.2-4.4">
+    <description>
+        calling differentially methylated regions from
+        bisulfite sequencing data
+    </description>
+    
+    <requirements>
+        <requirement type="package" version="0.2-4" >metilene</requirement>
+        <requirement type="binary">Rscript</requirement>
+        <requirement type="package" version="5.18.1" >perl</requirement>
+        <requirement type="package" version="3.2.1" >R</requirement>
+        <requirement type="package" version="2.24" >bedtools</requirement>
+        <requirement type="package" version="8.22">gnu_coreutils</requirement>
+    </requirements>
+    
+    <stdio>
+        <exit_code range=":-1"/>
+        <exit_code range="1:" />
+    </stdio>
+
+    <version_command>echo "0.2-4"</version_command>
+
+    <command><![CDATA[
+    ## =============== preprocessing
+    ##   generate metilene_g1_g2.input from multiple bedgraph files
+    metilene_input.pl 
+    --in1 $opt_in1
```

Comment:
I see. Of course there are workarounds like explicitely testing the
return status of the postprocessing :) -- but the metilene authors
should rather fix their tool.

From a different perspective, it would still be nice to be able to flag
single output files as good or bad. Obviously, one can decide to leave
an output file empty, but it seems like there is no fail condition (even
explicitly removing the output file does not help).

On 12/16/2015 02:54 PM, Björn Grüning wrote:
> In tools/metilene/metilene.xml
> <https://github.com/bgruening/galaxytools/pull/279#discussion_r47777355>:
> 
>> +        <requirement type="package" version="2.24" >bedtools</requirement>
>> +        <requirement type="package" version="8.22">gnu_coreutils</requirement>
>> +    </requirements>
>> +    
>> +    <stdio>
>> +        <exit_code range=":-1"/>
>> +        <exit_code range="1:" />
>> +    </stdio>
>> +
>> +    <version_command>echo "0.2-4"</version_command>
>> +
>> +    <command><![CDATA[
>> +    ## =============== preprocessing
>> +    ##   generate metilene_g1_g2.input from multiple bedgraph files
>> +    metilene_input.pl 
>> +    --in1 $opt_in1
> 
> No this is not possible as this is one job send to the cluster, it fails
> or it's gree :)
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bgruening/galaxytools/pull/279/files#r47777355>.
> 


-- 
Dr. Sebastian Will
Bioinformatics/IZBI
Härtelstrasse 16-18, Room 448.1
University Leipzig
D-04107 Leipzig

Tel.: +49 (0) 341 97 16689